index.js

```diff
@@ -44,9 +55,15 @@ class ExecutorQueue extends Executor {
      * @return {Promise}
      */
     _start(config) {
-        return this.breaker.runCommand('connect')
-            // Note: arguments to enqueue are [queue name, job type, array of args]
-            .then(() => this.breaker.runCommand('enqueue', this.buildQueue, 'start', [config]));
+        return this.queueBreaker.runCommand('connect')
+            // Store the config in redis
+            .then(() => this.redisBreaker.runCommand('hset', this.buildConfigTable,
+                config.buildId, JSON.stringify(config)))
+            // Note: arguments to enqueue are [queue name, job name, array of args]
+            .then(() => this.queueBreaker.runCommand('enqueue', this.buildQueue, 'start', [{
+                buildId: config.buildId,
+                annotations: config.annotations || ''
```

Comment:
Oh, you're right, that's going to cause possible problems for not enqueueing the same job twice.

Here we only have access to the parameters passed to `executor.start`. We could potentially add an optional `jobId` field to that configuration and use that.

Slightly related: I've realized that the built-in `jobLock` plugin is not going to work for us since it, like `queue.del`, looks for an exact parameter match. Since different builds will have different buildIds even if they're both the same job this means that we'll have to write our own plugin to handle that.